Modules/vCenterTls/vCenterTls.psm1

```diff
@@ -0,0 +1,691 @@
+
+    End
+    {
+    }
+}
+
+function ConvertTo-X509Certificate {
+    <#
+    .Synopsis
+        Converts PEM Base64 encoded string to an X509 Certificate
+    .DESCRIPTION
+        Converts PEM Base64 encoded string to an X509 Certificate
+    .PARAMETER CertificateData
+        PEM encoded certificate chain file
+    #>
+
+    [CmdletBinding(DefaultParameterSetName="All")]
+    
+    Param
+    (
+        [Parameter(Mandatory=$true,ValueFromPipeline=$true,Position=0)]
+        [String]
+        $CertificateData
+    )
+    
+    Begin{
+    }
+    Process
+    {
+        $BeginString = '-----BEGIN CERTIFICATE-----'
+        $EndString = '-----END CERTIFICATE-----'
+
+        foreach($Line in ($CertificateData -split '\n')) {
+            switch -Regex ($Line) {
+                $BeginString {
+                    #Create a new string when we see ---begin certificate---
+                    $Chain = New-Object System.Text.StringBuilder
+                }
+                $EndString {
+                    [System.Security.Cryptography.X509Certificates.X509Certificate2]([System.Convert]::FromBase64String($Chain.ToString()))
+                    $Chain.Clear() | Out-Null
+                }
+                default {
+                    $Chain.Append($Line) | Out-Null
+                }
+            }
+        }
+    }
+
+    End
+    {
+    }
+}
+
+function Set-NetCertficatePolicy {
+    <#
+    .Synopsis
+        Sets the System.Net.ServicePointManager Certificate Policy
+    .DESCRIPTION
+        Sets the System.Net.ServicePointManager Certificate Policy
+    .PARAMETER Policy
+        Default|TrustAll
+    .EXAMPLE
+        Set-CertificatePolicy -Policy TrustAll
+    .EXAMPLE
+        Set-CertificatePolicy -Policy Default
+    #>
+
+    [CmdletBinding(DefaultParameterSetName="All")]
+    
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateSet('TrustAll','Default')]
+        [String]
+        $Policy
+    )
+    
+    Begin{
+    }
+    Process
+    {
+        switch($Policy) {
+            'TrustAll' {
+                class TrustAllCertsPolicy : System.Net.ICertificatePolicy
+                {
+                    [bool] CheckValidationResult(
+                        [System.Net.ServicePoint] $a,
+                        [System.Security.Cryptography.X509Certificates.X509Certificate] $b,
+                        [System.Net.WebRequest] $c,
+                        [int] $d
+                    )
+                    {
+                        return $true
+                    }
+                }
+                [System.Net.ServicePointManager]::CertificatePolicy = [TrustAllCertsPolicy]::new()
+            }
+            'Default' {
+                [System.Net.ServicePointManager]::CertificatePolicy = $null
+            }
+            default{
+                [System.Net.ServicePointManager]::CertificatePolicy = $null
+            }
+        }
+    }
+    End
+    {
+    }
+}
+
+function Invoke-CisRest {
```

Comment:
Rename to Invoke-VCRestWithCISSession, or at least document what exactly CisRest means